polkadot/node/network/collator-protocol/src/validator_side/mod.rs

```diff
@@ -2086,3 +2109,51 @@ async fn handle_collation_fetch_response(
 	state.metrics.on_request(metrics_result);
 	result
 }
+
+// Sanity check the candidate descriptor version.
+fn descriptor_version_sanity_check(
+	descriptor: &CandidateDescriptorV2,
+	per_relay_parent: &PerRelayParent,
+) -> std::result::Result<(), SecondingError> {
+	if per_relay_parent.v2_receipts {
+		if let Some(core_index) = descriptor.core_index() {
+			if core_index != per_relay_parent.current_core {
+				return Err(SecondingError::InvalidCoreIndex(
+					core_index.0,
+					per_relay_parent.current_core.0,
+				))
+			}
+		} else {
+			return Err(SecondingError::InvalidReceiptVersion(
+				CandidateDescriptorVersion::V1,
+				CandidateDescriptorVersion::V2,
+			))
+		}
+
+		if let Some(session_index) = descriptor.session_index() {
+			if session_index != per_relay_parent.session_index {
+				return Err(SecondingError::InvalidSessionIndex(
+					session_index,
+					per_relay_parent.session_index,
+				))
+			}
+		} else {
+			return Err(SecondingError::InvalidReceiptVersion(
+				CandidateDescriptorVersion::V1,
+				CandidateDescriptorVersion::V2,
+			))
+		}
+
+		Ok(())
+	} else {
+		let descriptor_version = descriptor.version();
+		if descriptor_version != CandidateDescriptorVersion::V1 {
```

Comment:
This check is more robust if we  `== v2`.

Reply:
why? if the version is `Unknown`, this would pass. I don't think we want to allow such a candidate. do we?